### ISSUE TITLE
ksh.1 (again) - fix '|' implementation - hard to see - hard to find --

### ISSUE THREAD

**Issue by aabatpurdue** [att/ast#871]
**Description of problem:**
I did a `man path/ksh.1` and then searched for the **|** character.  Couldn't find even one.  Upon examination, I found that **\\(bv** was being used for **|**.  Changed them to an explicit **|**.  _groff_char(7)_ indicates that is the way to go.  There are also several **\\|** instances but I was unsure what they were being used for so I left them.  There are several **|\\||** instances that are displayed as **||** and I did not check to see if I could find them as **||** in the man page.

Fixed one typo also.  There is a second one in that typo's vicinity but, of course, I lost its position (:@{).

**Ksh version:**
September 3rd clone.

**How reproducible:**
n/a

**Steps to reproduce:**
1. `man path/ksh.1`
2.
3.

**Actual results:**
Could't find **|**.

**Expected results:**
Could find **|**.

**Additional info:**
Oops forgot to point to this patch [kshman2_patch.txt](https://github.com/att/ast/files/2381435/kshman2_patch.txt)

**Comment by krader1961** [comment]
> So obviously the ksh team made some ill-advised change.

I take that back. The raw man page in both ksh93u+ and ksh93v- (on which the current code in this project is based) are identical with respect to this issue. Running both files through `nroff -man` only yields three literal vertical bars. It's been so long since I've had to debug issues with roff man page source I had forgotten that `nroff -man /usr/local/share/man/man1/ksh.1` is not the same as `man /usr/local/share/man/man1/ksh.1`. When I do the latter all the expected vertical bars can be found.

So, @aabatpurdue, this appears to be a quirk of your system. Perhaps you are using a CJK or similar locale. But the ksh man page source is fundamentally the same between now and the last public release. And looking at both man page source files I see exactly the same discrepancy between the number of vertical bars I can find after running the file through `nroff -man` and just `man`. Thus I do not think we should apply your patch. At least not without a lot more discussion.

**Comment by krader1961** [comment]
Looking at the man page source it appears the three places I can see a literal vertical-bar in the `nroff -man src/cmd/ksh93/ksh.1` output are the only ones to use a literal ASCII vertical-bar. All the other places use `\bv`. So the question is whether those three places should be changed to `\bv` or all the `\bv` uses changed to a ASCII vertical-bar.

**Comment by aabatpurdue** [comment]
@krader1961 - Hmmm.   On Cygwin I just tried again using a July 27 clone and the `ksh.1.gz` that I installed from it. With both the file and the installed man page using `man`, I found just your three vertical bars.  The same with `nroff -mandoc` (<-- default macro package used by `man`) and `nroff -man` of the `./src/cmd/ksh93/ksh.1` file.  I also used `vim` to examine the raw roff file and found only three.  It was very interesting because I could see a vertical bar in `vim`s page but could not get `vim` to acknowledge its existence.

As far as I know, I have done nothing exotic.  Most all of everything is default.  I am using `mintty` with a 11pt consolas font.  TERM=xterm-256color.  Can you suggest anything else I might look for?  One problem may be that I do not have a specific locale set.  The `nroff` output is probably utf8 and the **\\(bv** would not be rendered as an ascii **|**.  Hey, I knew that if I talked long enough that I could find a solution.  If I set LC_ALL=C in the environment I can find all of the **|**.

Unfortunately, without the patch that still leaves a man page that will have similar problems if the user does not have the locale set properly.

And to get personal - punched cards?  You must be almost as old as I am (:-}).  I sure wasted a lot of paper and cardboard back in those days.

**Comment by krader1961** [comment]
> If I set LC_ALL=C in the environment I can find all of the |.

That implies there is a problem with the locale subsystem on your platform. As the C/POSIX locale should be the default locale is no locale environment vars (e.g., `LANG`, `LC_ALL`) are defined.

Yes, I'm a grey beard. I started programming in high school in 1977 and used both model 33 KSR teletype for learning BASIC then punched cards for FORTRAN. I was still working with punch cards (albeit not very much) in the early 1980's.

**Comment by aabatpurdue** [comment]
Shouldn't use this for interpersonal communication but don't have an email address for you.  I was a graduate student in Engineering Geology at Purdue in 1967 when one of my advisors suggested that I take a beginning FORTRAN programming course.  Boy, talk about love at first bite.  I then graduated to assembly language and that knowledge of assembly language is what eventually got me a job as a systems administrator (called a pud programmer back then) with the Research Computing group at Purdue.  I held that job in various guises from 1974 to 2012.  FWIW - I also got my MS degree.
